src/api/app/queries/workflow_runs_finder.rb

```diff
@@ -40,29 +40,11 @@ def with_generic_event_type(generic_event_type, request_action = nil)
       workflow_runs = workflow_runs.where("JSON_EXTRACT(request_payload, '$.action') = (?) OR JSON_EXTRACT(request_payload, '$.object_attributes.action') = (?)", request_action,
                                           request_action)
     end
-    workflow_runs
-  end
-
-  def with_event_source_name(event_source_name, request_action = nil)
-
-    query = "event_source_name"
-
-    # query = case generic_event_type
-    #         when 'tag_push'
-    #           "request_headers LIKE '%: Tag Push Hook%' OR JSON_EXTRACT(request_payload, '$.ref') LIKE '%refs/tags/%'"
-    #         when 'push'
-    #           "request_headers LIKE '%: Push Hook%' OR JSON_EXTRACT(request_payload, '$.ref') LIKE '%refs/heads/%'"
-    #         else
-    #           EVENT_TYPE_MAPPING[generic_event_type].map do |event_type|
-    #             "request_headers LIKE '%: #{event_type}%'"
-    #           end.join(' OR ')
-    #         end
 
-    workflow_runs = @relation.where(query)
-    if request_action && generic_event_type == 'pull_request'
-      workflow_runs = workflow_runs.where("JSON_EXTRACT(request_payload, '$.action') = (?) OR JSON_EXTRACT(request_payload, '$.object_attributes.action') = (?)", request_action,
-                                          request_action)
+    if event_source_name
+      workflow_runs = workflow_runs.where(event_source_name: event_source_name)
     end
```

Comment:
```suggestion
    return workflow_runs.where(event_source_name: event_source_name) if event_source_name
```

Reply:
This code has been moved to an independent method. Please have another look.